src/main/java/spring/intro/config/AppConfig.java

```diff
@@ -1,27 +1,35 @@
 package spring.intro.config;
 
-import java.util.Properties;
-import javax.sql.DataSource;
 import org.apache.commons.dbcp2.BasicDataSource;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
 import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
 import spring.intro.model.User;
 
+import javax.sql.DataSource;
+import java.util.Properties;
+
 @Configuration
+@PropertySource("classpath:db.properties")
 @ComponentScan(basePackages = {
         "spring.intro.service",
         "spring.intro.dao"
 })
 public class AppConfig {
+    @Autowired
+    private Environment env;
```

Comment:
What is better: setter, constructor, or field injection?

Reply:
Constructor is the best.